### PR TITLE
DW-4490 - Create `internet-ingress` VPC and related subnets

### PR DESCRIPTION
**Pull request by benjaminsherwood**
Also contains a non-working deployment of a reverse proxy, which is why there are some commented out areas.

### CODE REVIEW

File: locals.tf [hunk 1]
```diff
@@ -0,0 +1,11 @@
+locals {
+  reverse_proxy_ssmenabled = {
+    management-dev = "True"
+    management     = "False"
```
Comment (mattburgess) [inline locals.tf:4]:
```suggestion
    management-dev = true
    management     = false
```

File: vpc.tf [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  availability_zone = data.aws_availability_zones.available.names[count.index]
+  vpc_id            = module.vpc.vpc.id
+
+  tags = merge(
+    local.common_tags,
+    { Name = "reverse-proxy" }
+  )
+}
+
+resource "aws_subnet" "internet_proxy" {
```
Comment (mattburgess) [inline vpc.tf]:
```suggestion
resource "aws_subnet" "reverse_proxy" {
```

File: vpc.tf [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  availability_zone = data.aws_availability_zones.available.names[count.index]
+  vpc_id            = module.vpc.vpc.id
+
+  tags = merge(
+    local.common_tags,
+    { Name = "reverse-proxy" }
+  )
+}
+
+resource "aws_subnet" "reverse_proxy" {
```
Comment (mattburgess) [inline vpc.tf]:
```suggestion
resource "aws_subnet" "ssh_bastion" {
```

File: vpc.tf [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+  count             = length(data.aws_availability_zones.available.names)
+  cidr_block        = cidrsubnet(module.vpc.vpc.cidr_block, 4, count.index + length(aws_subnet.vpc_endpoint) + length(aws_subnet.internet_proxy))
```
Comment (mattburgess) [inline vpc.tf]:
```suggestion
  cidr_block        = cidrsubnet(module.vpc.vpc.cidr_block, 4, count.index + length(aws_subnet.vpc_endpoint) + length(aws_subnet.reverse_proxy))
```

File: vpc.tf [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+  availability_zone = data.aws_availability_zones.available.names[count.index]
+  vpc_id            = module.vpc.vpc.id
+
+  tags = merge(
+    local.common_tags,
+    { Name = "reverse-proxy" }
```
Comment (mattburgess) [inline vpc.tf]:
```suggestion
    { Name = "ssh-bastion" }
```

File: locals.tf (hunk 1 above)
Comment (benjaminsherwood) [reply locals.tf:4]: Tag not boolean

**Review by mattburgess — APPROVED** [review]

**Pull request merged**